api.js

```diff
@@ -140,12 +147,10 @@ RosterAPI.prototype.request = function(path, opts) {
         this.dataTableAttach();
 
     if (!('headers' in opts)) opts.headers = {};
-
-    var credentials = this.credentials;
-    if ('credentials' in opts) credentials = opts.credentials;
+    var credentials = opts.credentials || this.credentials;
 
     if (credentials) {
-        opts.headers['Authorization'] = "ApiKey " + credentials.key;
+        opts.headers['Authorization'] = "ApiKey " + this.key;
```

Comment:
@gregstuderoaf to me it was a little confusing to have the key stored in 'this.key' and in 'credentials.key' 

Plus, this was causing the first-time-failed-authClient bug. Because "credentials.key" was undefined, but "this.key" WAS defined by the call to "dataTableAttach" on line 147. 

So I thought  it easiest to remove the "credentials.key" field.

Reply:
Yeah I agree, my thoughts were really that there may be different API keys in the future for different user types or something, but I'm not sure that's right either honestly.

Reply:
Oh, but do we still *store* the key in credentials.key after this patch?  We should stop doing that at the same time if we never access it there.

Reply:
hmmm I don't think credentials.key is ever referenced or stored -- but do you see something I don't see? @gregstuderoaf

Reply:
Yup it's not.  Cool.